### PR TITLE
Hsc update - Send button + dropdowns

### LINKED ISSUE #880: GUI: LSL Sends Corrupt Data When Filters On
## Problem

When Filters are toggled on during LSL Networking, the data sent out is corrupt.
This is only noticeable when using Cyton, and NOT when using Synthetic data.
I tested this extensively, many times using different filters.

## Expected 

Expect that data sent over LSL is exactly as what is visualized on the GUI, when Filters are toggled ON.
Rather, currently, the data is completely corrupt.

## Operating System and Version

Windows

## GUI Version

5.0.0

## Running standalone app

Running the downloaded App.

## Type of OpenBCI Board

Cyton.

## Are you using a WiFi Shield?

No, using dongle.

### PR DESCRIPTION
**Pull request by retiutut**
Update Hardware Settings Controller:
- [x] Send button
- [x] Swap out old buttons for cp5 dropdowns
- [x] Lock dropdowns when channel is off
- [x] **Apply Daniel's suggestion https://github.com/OpenBCI/OpenBCI_GUI/pull/879#issuecomment-697994494**
- [x] Short channels automatically when turning them off (like before with buttons)
- [x] User has unchanged settings pop-up  https://github.com/OpenBCI/OpenBCI_GUI/pull/879#issuecomment-704356756
- [x] Require users to Send or Revert hardware settings before closing controller
______

- [x] Fix LSL TimeSeries Filtered data output Fixes #880 
- [x] Update changelog and bump version

### CODE REVIEW

File: OpenBCI_GUI/ADS1299SettingsBoard.pde [hunk 1]
```diff
@@ -326,10 +253,30 @@ class ADS1299Settings {
...
+    public boolean[] commitAll() {
+        int numChan = board.getNumEXGChannels();
+        boolean[] success = new boolean[numChan];
+        for (int i=0; i<numChan; i++) {
+            success[i] = commit(i);
         }
+        return success;
+    }
+
+    public void saveDefaultValues() {
```
Comment (Andrey1994) [inline OpenBCI_GUI/ADS1299SettingsBoard.pde:266]: Is it really default values or current\last commited values? As I understand its more like current values and in this case we should rename it

**Review by Andrey1994 — APPROVED** [review]

**Comment by daniellasry** [comment]
I'm seeing numbers printed on launch:
```
OpenBCI_GUI::Setup: Setup is complete!
29.274
36.516
39.984
43.350002
46.716
50.184
53.652
57.018
60.384
63.852
67.217995
70.584
74.052
77.622
80.98801
84.354
87.72
91.188
94.656
98.022
101.49
105.06
108.425995
111.792
115.259995
118.626
122.094
125.562
128.928
132.39601
135.864
139.332
142.698
146.064
149.43001
152.89801
156.366
159.732
163.302
166.668
170.136
173.502
177.072
180.54001
183.804
187.17001
190.638
194.106
197.574
201.04199
204.51
207.87599
211.14
214.608
217.872
221.238
224.808
228.174
231.642
235.11
238.37401
241.944
245.31
248.778
252.246
255.51
```

**Comment by retiutut** [comment]
Debugging happens to be on at the moment, the numbers are related to the fade-in animation, as sort of a drawing timer check.

**Comment by daniellasry** [comment]
It looks like the red highlight means "these channel's settings failed to apply"

What if we expanded it to mean "this channel has unapplied changes" or "this channel is out of sync with the board"? My thinking is that it would clarify to the user that they have unapplied changes, if they try to close the hardware settings witout sending.

The difference would be that we would highlight the channel as soon as the user changes something, and only remove the highlight if that channels' settings were successfully sent. 

Ideally, if a channel was unchanged, we should not highlight it in red even if it fails to send, because that channel is not out of sync with the board.

**Comment by retiutut** [comment]
@daniellasry Sounds like a better idea! Needed some clarification on the UX anyways.

**Review by retiutut — COMMENTED** [review]
Should work as expected!

**Comment by daniellasry** [comment]
Looking good, there's one thing missing:
If i change a setting without sending and exit Hardware Settings, nothing happens.
It seems like when i go back into HW settings, it remembers my state (the channel is still highlighted).

Should we warn the user they have unsent changes? What did we settle on?

**Comment by retiutut** [comment]
@daniellasry Thanks for catching that. I will make the change.

**Comment by retiutut** [comment]
```
public void setIsVisible (boolean v) {
        
        //Check if there are unapplied settings when trying to close Hardware Settings Controller
        if (!v) {
            boolean allChannelsInSync = true;

            for (int i = 0; i < hasUnappliedChanges.length; i++) {
                if (hasUnappliedChanges[i]) {
                    allChannelsInSync = false;
                }
            }

            if (!allChannelsInSync) {
                PopupMessage msg = new PopupMessage("Info", "Highlighted channels have unapplied Hardware Settings. Please press \"Send\" button to sync with board or revert settings.");
                return;
            }
        }

        isVisible = v;
    }
```

**Review by daniellasry — APPROVED** [review]
👍

**Pull request merged**